src/Microsoft.AspNet.Razor/Chunks/ChunkTreeBuilder.cs

```diff
@@ -69,22 +70,39 @@ public void AddRemoveTagHelperChunk(string lookupText, SyntaxTreeNode associatio
 
         public void AddLiteralChunk(string literal, SyntaxTreeNode association)
         {
-            // If the previous chunk was also a LiteralChunk, append the content of the current node to the previous one.
-            var literalChunk = _lastChunk as LiteralChunk;
-            if (literalChunk != null)
+            ParentLiteralChunk parentLiteralChunk;
```

Comment:
For a thought exercise I rewrote the following to normalize the Adding of literal chunks. Torn as to which one is more clear. Therefore this is probably fine as is. Here's it rewritten in-case you feel differently:

```C#
var parentLiteralChunk = _lastChunk as ParentLiteralChunk;

// We try to join literal chunks where possible, so that we have fewer 'writes' in the generated code.
if (parentLiteralChunk == null && _lastChunk is LiteralChunk)
{
    Debug.Assert(Current.Children[Current.Children.Count - 1] == _lastChunk);
    Current.Children.RemoveAt(Current.Children.Count - 1);

    parentLiteralChunk = new ParentLiteralChunk
    {
        Start = _lastChunk.Start,
    };
    parentLiteralChunk.Children.Add(_lastChunk);
}

if (parentLiteralChunk != null)
{
    parentLiteralChunk.Children.Add(new LiteralChunk
    {
        Association = association,
        Start = association.Start,
        Text = literal,
    });

    _lastChunk = parentLiteralChunk;
}
else
{
    AddChunk(new LiteralChunk
    {
        Text = literal,
    }, association);
}
```